PsfRuntime/CreateProcessHook.cpp

```diff
@@ -156,77 +216,106 @@ BOOL WINAPI CreateProcessFixup(
+                    try
                     {
-                        static const auto altDirPathToPsfRuntime = narrow(dentry.path().c_str());
+                        if (dentry.path().filename().compare(wtargetDllName) == 0)
+                        {
+                            static const auto altDirPathToPsfRuntime = narrow(dentry.path().c_str());
 #if _DEBUG
-                        Log("\tFound match as %ls", dentry.path().c_str());
+                            Log("\tFound match as %ls", dentry.path().c_str());
 #endif
-                        targetDll = altDirPathToPsfRuntime.c_str();
-                        break;
+                            targetDllPath = altDirPathToPsfRuntime.c_str();
+                            break;
+                        }
+                    }
+                    catch (...)
+                    {
+                        Log("Non-fatal error enumerating directories while looking for PsfRuntime.");
                     }
                 }
-                catch (...)
-                {
-                    Log("Non-fatal error enumerating directories while looking for PsfRuntime.");
-                }
+
             }
         }
-      
-        Log("\tAttempt injection into %d using %s", processInformation->dwProcessId, targetDll);
-        if (!::DetourUpdateProcessWithDll(processInformation->hProcess, &targetDll, 1))
+
+        if (targetDllPath != NULL)
         {
-            Log("\t%s not found at target folder, try PsfRunDll.", targetDll);
-            // We failed to detour the created process. Assume that it the failure was due to an architecture mis-match
-            // and try the launch using PsfRunDll
-            if (!::DetourProcessViaHelperDllsW(processInformation->dwProcessId, 1, &targetDll, CreateProcessWithPsfRunDll))
+            Log("\tAttempt injection into %d using %s", processInformation->dwProcessId, targetDllPath);
+            if (!::DetourUpdateProcessWithDll(processInformation->hProcess, &targetDllPath, 1))
             {
-                // Could not detour the target process, so return failure
-                auto err = ::GetLastError();
-                Log("\tUnable to inject %ls into PID=%d err=0x%x\n", psf::runtime_dll_name, processInformation->dwProcessId, err);
-                ::TerminateProcess(processInformation->hProcess, ~0u);
-                ::CloseHandle(processInformation->hProcess);
-                ::CloseHandle(processInformation->hThread);
-
-                ::SetLastError(err);
-                return FALSE;
+                Log("\t%ls unable to inject, skipping.", targetDllPath);
+                // We failed to detour the created process. Assume that it the failure was due to an architecture mis-match
+                // and try the launch using PsfRunDll
+                //if (!::DetourProcessViaHelperDllsW(processInformation->dwProcessId, 1, &targetDllPath, CreateProcessWithPsfRunDll))

```

Comment:
Remove this block of commented out code.

Reply:
Done.